File: dashboards/ruby/routes/messages.rb
# frozen_string_literal: true

# Messages routes — list, send

get '/messages' do
  require_auth!
  res = api_get('/engine/messages?orgId=default')
  @messages = res[:status] == 200 ? (res[:body]['messages'] || res[:body]) : []
  @messages = @messages.is_a?(Array) ? @messages : []
  erb :messages
end

post '/messages/send' do
  require_auth!
  res = api_post('/engine/messages', {
    to: params[:to],
    subject: params[:subject],
    body: params[:body],
    orgId: 'default'
  })
  if res[:status] < 300
    set_flash('Message sent successfully', 'success')
  else
    set_flash(res[:body].is_a?(Hash) ? (res[:body]['error'] || 'Failed to send message') : 'Failed to send message', 'danger')
  end
  redirect '/messages'
end

File: dashboards/python/routes/messages.py
"""
Messages routes: list and send messages.
"""

from flask import Blueprint, render_template, request, session, redirect, url_for, flash
from functools import wraps
from utils.api import api_request

messages_bp = Blueprint('messages_bp', __name__)


def login_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        if 'token' not in session:
            return redirect(url_for('auth_bp.login'))
        return f(*args, **kwargs)
    return decorated


@messages_bp.route('/messages')
@login_required
def messages():
    data = api_request('GET', '/engine/messages?orgId=default')
    return render_template(
        'messages.html',
        active_page='messages',
        user=session.get('user', {}),
        messages=data.get('messages', []),
    )


@messages_bp.route('/messages/send', methods=['POST'])
@login_required
def send_message():
    body = {
        'type': request.form.get('type', 'email'),
        'from': request.form['from_addr'],
        'to': request.form['to_addr'],
        'subject': request.form['subject'],
        'body': request.form.get('body', ''),
        'priority': request.form.get('priority', 'normal'),
    }
    result = api_request('POST', '/engine/messages', body)
    flash('Message sent!' if 'id' in result else result.get('error', 'Failed'))
    return redirect(url_for('messages_bp.messages'))
